pennylane/templates/subroutines/qdrift.py

```diff
@@ -144,6 +144,17 @@ def my_circ(time):
 
     """
 
+    def _flatten(self):
+        h = self.hyperparameters["base"]
+        hashable_hyperparameters = tuple(
+            (key, value) for key, value in self.hyperparameters.items() if key != "base"
+        )
+        return (h, self.data[-1]), hashable_hyperparameters
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        return cls(*data, **dict(metadata))
```

Comment:
This is a unflatten/flatten structure consistent with other, similar operations. This includes `ApproxTimeEvolution`, `Trotter` and `CommutingEvolution`.